volatility3/framework/symbols/linux/extensions/__init__.py

```
@@ -3182,10 +3211,38 @@ def get_name(self) -> Optional[str]:
         # if everything else failed...
         if hasattr(self, "mattr"):
             try:
-                return utility.pointer_to_string(self.mattr.attr.name, count=32)
+                return utility.pointer_to_string(
+                    self.mattr.attr.name, count=linux_constants.ATTRIBUTE_NAME_MAX_SIZE
+                )
             except exceptions.InvalidAddressException:
                 vollog.debug(
                     f"Unresolvable name for for section at {self.vol.offset:#x}"
                 )
 
         return None
+
+
+class bin_attribute(objects.StructType):
+    def get_name(self) -> Optional[str]:
+        """
+        Performs extraction of the bin_attribute name
+        """
+        if hasattr(self, "attr"):
```

Do we know what circumstances this might *not* have an `attr` attribute?  Is there a particular kernel version, or some other indicator we could put in there for when this might happen?

---

`bin_attribute` should always have this member, I think I'll remove the hasattr call. If it breaks in a new kernel version, at least it won't silently return empty strings and we'll be able to update the method.